### PR TITLE
NF: ConfigManager.obtain() for (interactive) config query (fixes gh-836)

### PR DESCRIPTION
**Pull request by mih**

### CODE REVIEW

**Comment by mih** [comment]
Fighting with the buildbot setup...

**Comment by mih** [comment]
Nice, so now the buildbots are happy, but travis no longer....

**Comment by mih** [comment]
They seem to have just died ...

**Comment by mih** [comment]
@yarikoptic @bpoldrack please take a look -- speedy merge is targeted

**Comment by bpoldrack** [comment]
I like it. No objections. May be just a little more doc on how the ui definitions are supposed to look like. So in half a year we don't need to track down the code to figure out, what are valid values for "destination" and what is the meaning of which keyword ...

File: datalad/support/dsconfig.py [hunk 1]
```diff
@@ -141,6 +142,124 @@ def reload(self):
...
+                    "not preconfigured, no default".format(var))
+            # XXX maybe we should return default here, even it was returned
+            # from the UI -- if that is even possible
+
+        # execute type conversion before storing to check that we got
+        # something that looks like what we want
+        try:
+            value = valtype(_value)
+        except Exception as e:
+            raise ValueError(
```
Comment (yarikoptic) [inline datalad/support/dsconfig.py]: `assert_raises` is very cool ;)

**Comment by mih** [comment]
For the type conversion this assert is easy to test, but I cannot think of a way to reach line 237. Any idea?

**Comment by mih** [comment]
Ready to merge from my POV

**Comment by mih** [comment]
MERGE!

**Pull request merged**